pkg/devicerepository/store/bleve/store.go

```diff
@@ -179,8 +183,60 @@ func (s *bleveStore) GetModels(req store.GetModelsRequest) (*store.GetModelsResp
 }
 
 // GetTemplate retrieves an end device template for an end device definition.
-func (s *bleveStore) GetTemplate(ids *ttnpb.EndDeviceVersionIdentifiers) (*ttnpb.EndDeviceTemplate, error) {
-	return s.store.GetTemplate(ids)
+func (s *bleveStore) GetTemplate(req *ttnpb.GetTemplateRequest, _ *store.EndDeviceProfile) (*ttnpb.EndDeviceTemplate, error) {
+	var (
+		profile             *store.EndDeviceProfile
+		endDeviceProfileIds = req.GetEndDeviceProfileIds()
+	)
+	// Attempt to fetch the end device profile that matches the identifiers.
+	if endDeviceProfileIds != nil {
+		documentTypeQuery := bleve.NewTermQuery(profileDocumentType)
+		documentTypeQuery.SetField("Type")
+		queries := []query.Query{documentTypeQuery}
+		if q := endDeviceProfileIds.VendorId; q != 0 {
+			query := bleve.NewTermQuery(strconv.Itoa(int(q)))
+			query.SetField("VendorID")
+			queries = append(queries, query)
+		}
+		query := bleve.NewTermQuery(strconv.Itoa(int(endDeviceProfileIds.VendorProfileId)))
+		query.SetField("VendorProfileID")
+		queries = append(queries, query)
+
+		searchRequest := bleve.NewSearchRequest(bleve.NewConjunctionQuery(queries...))
+		searchRequest.Fields = []string{"ProfileJSON", "BrandID"}
+		result, err := s.index.Search(searchRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(result.Hits) == 0 {
+			return nil, errEndDeviceProfileNotFound.WithAttributes(
+				"vendor_id", endDeviceProfileIds.VendorId,
+				"vendor_profile_id", endDeviceProfileIds.VendorProfileId)
+		}
+
+		if len(result.Hits) != 1 {
+			// There can only be one profile for a given tuple.
+			return nil, errInvalidNumberOfProfiles.New()
```

Comment:
I don't think this is validated at the moment

@Jaime-Trinidad please file an issue to validate this

Reply:
https://github.com/TheThingsNetwork/lorawan-devices/issues/405